src/core/thread/network_data.hpp

```diff
@@ -353,6 +353,21 @@ class NetworkData: public ThreadNetifLocator
         kDataResubmitDelay = 300000,  ///< DATA_RESUBMIT_DELAY (miliseconds)
     };
 
+    class NetworkDataIterator
+    {
+    public:
+        NetworkDataIterator(otNetworkDataIterator *aIterator):
+            mIteratorBuffer(reinterpret_cast<uint8_t *>(aIterator)) { }
+
+        uint8_t GetTlvsIndex(void) const { return mIteratorBuffer[0]; }
+        uint8_t GetEntryIndex(void) const { return mIteratorBuffer[1]; }
+        void SetTlvsIndex(uint8_t aIndex) { mIteratorBuffer[0] = aIndex; }
+        void SetEntryIndex(uint8_t aIndex) { mIteratorBuffer[1] = aIndex; }
+
+    private:
+        uint8_t *mIteratorBuffer;
```

Comment:
Not with the model I am using here. We are not type-casting the `otNetworkInterator` to this class type, instead passing it in as a pointer in constructor.